rfd/0236-access-list-preset.md

```diff
@@ -0,0 +1,263 @@
+---
+author: Lisa Kim (lisa@goteleport.com)
+state: draft
+---
+
+# RFD 0236 - Access Lists Created with Presets
+
+# Required Approvers
+
+- Engineering: @r0mant || @smallinsky (marek) && @kopiczko (pavel)
+- Product: @r0mant
+
+## What
+
+Access lists allows an admin to grant additional roles (and traits) to a group of users on a long lived basis. Currently, to define grants for an access list, admins must select roles that already exists. This requires the admins to know what a role is and know how to create or update a role to customize access.
+
+This RFD proposes creating different guided access lists provided by Teleports web app, where it removes the need for admins to know about roles.
+
+The guide can also be used as a Terraform access list script generator.
+
+## Why
+
+Improves access list usability especially for day one users. Guided access list allows an admin to focus on users and what resources users should have access to. It removes the need for an admin to learn how to create/update roles and removes how roles have any relation to an access list because Teleport will do it for them.
+
+### Presets
+
+Presets will describe the type of guides that admins can use. It describes what kinds of actions that Teleport will take. More info on preset in user stories below.
+
+### User story: As an admin, I want to grant a select group of users (local or Okta imported) long term access to resources
+
+The admin can create an access list using the preset `long-term`.
+
+Preset `long-term` represents an access list that grants members long lived access to Teleport resources. Owners can review access requests. This access list is pretty similar to how access list works now (non-integrated types).
+
+All admin needs to do for this preset is to define the metadata, owners, and members of the access list and define the access to resources (e.g. node_labels and node logins) and Teleport will create the following type of roles and create/modify the access list accordingly:
+
+- `access`: Role(s) related to allowing access to resources. The access specs are determined by admin input. These roles are directly assigned to members of this access list (as member grants).
+- `requester`: A role that allows requesting for the resources defined in the `access` roles. This role is not automatically assigned to any user and is a role an admin can optionally assign to any user not assigned as a member to this access list.
+- `reviewer`: A role that allows reviewing access requests to resources defined in the "access" roles. This role is directly assigned to owners of this access list (as owner grants).
```

Comment:
Why are we creating a `requester` and `reviewer` roles for long-term access? Is it for promotions?